src/field/pokemon.ts

```diff
@@ -3338,6 +3336,8 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
       amount = 0;
     }
 
+    const ignoreDynamaxReduction = [HitResult.ONE_HIT_KO, HitResult.SELF_KO].includes(result) ? true : false;
```

Comment:
array.includes() already returns a boolean value
```suggestion
    const ignoreDynamaxReduction = [HitResult.ONE_HIT_KO, HitResult.SELF_KO].includes(result);
```

Reply:
Whoops, you're right.